webapp/app.py

```diff
@@ -198,7 +198,11 @@ def bad_request_error(error):
 
 @app.errorhandler(SecurityAPIError)
 def security_api_error(error):
-    return flask.render_template("security-error.html"), 500
+    message = error.response.json()
+    if error.response.status_code == 404:
+        return flask.render_template("security-error-404.html", message=message), 404
```

Comment:
You're certainly right that we need to catch 404s here. But I don't see the need to create a separate template, this just means we have more different templates to maintain, when the actual 404 page you've added is pretty similar.

I'm not sure if it's helpful to show the "CVE-xxxx does not exist" message, or if a normal 404 page will suffice. It's up to you. But if you do still want to pass through that message, you can still do it in a generic way without creating an extra template. Change this code to:

```suggestion
    if error.response.status_code == 404:
        flask.abort(404, message)
```

And then change the general `404.html` template to:

``` html
<p class="p-heading--4">{{ message or 'Sorry, we couldn't find that page.' }}</p>
```